### PR TITLE
variant: Add a `bytes()` method.

### PR DESCRIPTION
**Pull request by cgwalters**
Similarly to how we have a `str()` method which avoids copying
string data just to read it, add a method to access a byte array
directly.  Right now, dealing with GVariant type `ay` is extremely
expensive because we'll end up heap allocating a variant for *each
byte* whether one does `let data: Vec<u8> = v.get()` or iterates.

Note that following previous design discussion, this API returns
`Result` and not `Option`.

### CODE REVIEW

File: glib/src/variant.rs [hunk 1]
```diff
@@ -290,6 +290,28 @@ impl Variant {
         }
     }
 
+    /// Tries to extract a `&[u8]` from a variant of type `ay`.
```
Comment (GuillaumeGomez) [inline glib/src/variant.rs:293]: What does `ay` mean?

**Review by GuillaumeGomez — APPROVED** [review]

File: glib/src/variant.rs (hunk 1 above)
Comment (cgwalters) [reply glib/src/variant.rs:293]: https://gitlab.gnome.org/GNOME/glib/-/blob/be4b8cbe96ecd8d7d738fb9c80729285123ecdf3/glib/gvarianttype.c#L125

Comment (GuillaumeGomez) [reply glib/src/variant.rs:293]: Maybe write it in full text in the documentation then. Not everyone is a gnome developer (myself included).

Comment (cgwalters) [reply glib/src/variant.rs]: Done.

File: glib/src/variant.rs [hunk 2]
```diff
@@ -34,6 +34,14 @@
 //! assert_eq!(hello.str(), Some("Hello!"));
 //! assert_eq!(num.str(), None);
 //!
+//! // `bytes` tries to borrow a byte array (GVariant type `ay`).
```
Comment (GuillaumeGomez) [inline glib/src/variant.rs:37]: Please add the explanation here as well.

**Review by GuillaumeGomez — APPROVED** [review]

File: glib/src/variant.rs [hunk 3]
```diff
@@ -290,6 +300,28 @@ impl Variant {
         }
     }
 
+    /// Tries to extract a `&[u8]` from a variant of type `ay` (array of bytes).
+    ///
+    /// Returns an error if the type is not `ay`.
+    pub fn bytes(&self) -> Result<&[u8], VariantTypeMismatchError> {
```
Comment (sdroege) [inline glib/src/variant.rs:306]: Should really update all the other `Option`s to `Result`s :)

**Review by sdroege — APPROVED** [review]

File: glib/src/variant.rs (hunk 3 above)
Comment (cgwalters) [reply glib/src/variant.rs:306]: Yeah, I'd just also like to use this stuff on 0.14 so will try to do a larger pass for "0.15 API changes" at some point later.

**Pull request merged**